Source/WebCore/html/HTMLImageElement.cpp

```diff
@@ -486,8 +499,11 @@ void HTMLImageElement::removedFromAncestor(RemovalType removalType, ContainerNod
         selectImageSource(RelevantMutation::Yes);
     }
 
-    m_form = nullptr;
     HTMLElement::removedFromAncestor(removalType, oldParentOfRemovedTree);
+
+    // Do not rely on rootNode() because IsInTreeScope can be outdated.
+    if (m_form && &asHTMLElement().traverseToRootNode() != &m_form->traverseToRootNode())
```

Comment:
Unlike the place where we got this from in `FormAssociatedElement`, we should not need to call `asHTMLElement()` here. Just directly calling `traverseToRootNode()` should work.

Wish we could share a bit more code, but not such a big deal I guess.

Reply:
Good point. Fixed.

Reply:
> Wish we could share a bit more code, but not such a big deal I guess.

My plan is to do a bunch of renames & refactoring once this PR gets landed.